src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -654,18 +673,87 @@ def _map_line_buffers(
         point_buffer.append(indent_point)
 
         if not indent_point.is_line_break:
+            # If it's not a line break, we should still check whether it's
```

Comment:
See my question about #4234 and #4328. The latter issue, at least, seems like it may involve the same logic (line break / dedent interaction).

Reply:
yep - agreed. Thanks for updating the linked issues 👍